Alloy/Encoders/BoundingBoxesRenderer.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  BoundingBoxesRenderer.swift
+//  Alloy
+//
+//  Created by Eugene Bokhan on 22/04/2019.
+//
+
+import Metal
+import simd
+
+@available(iOS 11.3, tvOS 11.3, macOS 10.13, *)
+final public class BoundingBoxesRenderer {
```

Comment:
I really think this is over-complexity. Why not declare a cumulative index buffer for rectangle and render it in a single draw call?